aws_metadata/libraries/aws_metadata.rb

```diff
@@ -28,15 +26,17 @@ def self.get_aws_region
 
   def self.get_aws_vpc_id
     c = Chef::HTTP.new('http://169.254.169.254')
-    interfaces = c.get('/2016-09-02/meta-data/network/interfaces/macs/')
+    v2_token = c.put("/latest/api/token", nil, { 'X-aws-ec2-metadata-token-ttl-seconds': "120" })
+    interfaces = c.get('/2016-09-02/meta-data/network/interfaces/macs/', { 'X-aws-ec2-metadata-token' => v2_token })
     interfaces = interfaces.split("\n").map { |interface| interface.chomp("/") }
-    c.get("/2016-09-02/meta-data/network/interfaces/macs/#{interfaces.fetch(0)}/vpc-id")
+    c.get("/2016-09-02/meta-data/network/interfaces/macs/#{interfaces.fetch(0)}/vpc-id", { 'X-aws-ec2-metadata-token' => v2_token })
   end
 
   def self.get_aws_vpc_cidr
     c = Chef::HTTP.new('http://169.254.169.254')
-    interfaces = c.get('/2016-09-02/meta-data/network/interfaces/macs/')
+    v2_token = c.put("/latest/api/token", nil, { 'X-aws-ec2-metadata-token-ttl-seconds': "120" })
+    interfaces = c.get('/2016-09-02/meta-data/network/interfaces/macs/', { 'X-aws-ec2-metadata-token' => v2_token })
     interfaces = interfaces.split("\n").map { |interface| interface.chomp("/") }
-    c.get("/2016-09-02/meta-data/network/interfaces/macs/#{interfaces.fetch(0)}/vpc-ipv4-cidr-block")
+    c.get("/2016-09-02/meta-data/network/interfaces/macs/#{interfaces.fetch(0)}/vpc-ipv4-cidr-block", { 'X-aws-ec2-metadata-token' => v2_token })
```

Comment:
Just a nitpick, perhaps it's worth creating a method `v2_token_headers` and using that method for each of the header configs above.